src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/time.html.twig

```diff
@@ -142,422 +139,70 @@
-                });
-            };
-
-            this.getThreshold = function() {
-                var threshold = Sfjs.getPreference(_storagePrefix + 'threshold');
-
-                if (null === threshold) {
-                    return _threshold;
-                }
-
-                _threshold = parseInt(threshold);
-
-                return _threshold;
-            };
-
-            this.setThreshold = function(threshold)
-            {
-                _threshold = threshold;
-
-                Sfjs.setPreference(_storagePrefix + 'threshold', threshold);
-
-                return this;
-            };
-        }
-
-        function canvasAutoUpdateOnResizeAndSubmit(e) {
-            e.preventDefault();
-            canvasManager.drawAll();
-        }
-
-        function canvasAutoUpdateOnThresholdChange(e) {
-            canvasManager
-                .setThreshold(query('input[name="threshold"]').value)
-                .drawAll();
-        }
-
-        var requests_data = {
-            "max": {{ "%F"|format(collector.events.__section__.endtime) }},
-            "requests": [
-{{ helper.dump_request_data(token, collector.events, collector.events.__section__.origin) }}
-
-{% if profile.children|length %}
-                ,
-{% for child in profile.children %}
-{{ helper.dump_request_data(child.token, child.getcollector('time').events, collector.events.__section__.origin) }}{{ loop.last ? '' : ',' }}
-{% endfor %}
-{% endif %}
-            ]
-        };
-
-        var canvasManager = new CanvasManager(requests_data.requests, requests_data.max);
-
-        query('input[name="threshold"]').value = canvasManager.getThreshold();
-        canvasManager.drawAll();
-
-        // Update the colors of legends.
-        var timelineLegends = document.querySelectorAll('.sf-profiler-timeline > .legends > span[data-color]');
-
-        for (var i = 0; i < timelineLegends.length; ++i) {
-            var timelineLegend = timelineLegends[i];
-
-            timelineLegend.style.borderLeftColor = timelineLegend.getAttribute('data-color');
-        }
-
-        // Bind event handlers
-        var elementTimelineControl = query('#timeline-control'),
-            elementThresholdControl = query('input[name="threshold"]');
-
-        window.onresize = canvasAutoUpdateOnResizeAndSubmit;
-        elementTimelineControl.onsubmit = canvasAutoUpdateOnResizeAndSubmit;
-
-        elementThresholdControl.onclick = canvasAutoUpdateOnThresholdChange;
-        elementThresholdControl.onchange = canvasAutoUpdateOnThresholdChange;
-        elementThresholdControl.onkeyup = canvasAutoUpdateOnThresholdChange;
-
-        window.setTimeout(function() {
-            canvasAutoUpdateOnThresholdChange(null);
-        }, 50);
-
-    //]]>{% endautoescape %}</script>
+<style type="text/css">
+{% include '@WebProfiler/Collector/time.css.twig' with classnames %}
+</style>
+<script>
+{% include '@WebProfiler/Collector/time.js.twig' %}
```

Comment:
As there is strictly no Twig markup in this file, I suggest to remove the `.twig` extension. This way, tooling will know it is a JS file and treat it that way (for instance, Github currently does not highlight it as JS but as HTML, which of course fails)

Reply:
Fixed.